tests/delegated_scanning_test.go

```diff
@@ -177,6 +177,24 @@ func (ts *DelegatedScanningSuite) SetupSuite() {
 	// must have an active connection for this check to succeed, so wait for that connection.
 	ts.waitForHealthyCentralSensorConn()
 
+	matcherReady := func() error {
+		notReadyMessage := "initial load for the vulnerability store is in progress"
+		conn := centralgrpc.GRPCConnectionToCentral(t)
+		service := v1.NewImageServiceClient(conn)
+		_, err := service.ScanImage(ctx, &v1.ScanImageRequest{
+			ImageName: ts.ubi9Image.TagRef(),
+			Force:     true,
+			Cluster:   ts.remoteCluster.GetId(),
+		})
+
+		if err != nil && strings.Contains(err.Error(), notReadyMessage) {
+			return err
+		}
+		return nil
+	}
```

Comment:
Why did you move away from `SCANNER_V4_MATCHER_READINESS`?

Reply:
That discussion is related to on-prem customers. My question was for these tests. Do we need to also enable `readiness=vulnerability` for customers to have it enabled here? For example, in perf & scale tests [we are using readiness=vulnerability](https://github.com/openshift/release/pull/58792/files#diff-20b1d74a99d7d40882044ed2785fabfe2f9e21a90b8098dcba6a2ee663292d54R60).

Reply:
Thinking about it again, it does seem like just looking for the readiness flag instead of polling the API would be simpler, and we already validated that changing the readiness setting to `vulnerabilities` doesn't cause any other regressions (IIRC).  So perhaps it would be better to set it.

The issue now is that there is no way (afaict) to set it without changing the default in the product.  I don't see `SCANNER_V4_MATCHER_READINESS` referenced in the code anywhere (is that setting in CI nightlies actually doing anything?), though maybe I'm missing it somehow.  And the option in the matcher config file is not part of the Helm template, so I don't believe it can be configured as an installation parameter.

We could add it to be configurable in this PR if we want to go that route.

Reply:
So the answer is either:

    --set customize.scanner-v4-matcher.envVars.SCANNER_V4_MATCHER_READINESS=vulnerability

on the command line, or in `values.yaml`:

    customize:
      scanner-v4-matcher:
        envVars:
          SCANNER_V4_MATCHER_READINESS: vulnerability

But I guess the real question is how to apply this in CI.  The test suite I'm working on here uses the operator to deploy, so I need to set it in the `Central` custom resource, which has its own `customize.envVars` option.  There is a custom way to set these env vars in the e2e shell scripts.  I made the change in [b5776c5](https://github.com/stackrox/stackrox/pull/14956/commits/b5776c5e322aae46dba66120acef99ee04c5c3e4)